File: Concesionario/src/gestorAplicacion/activos/Articulo.java
package gestorAplicacion.activos;

import java.io.Serializable;

public class Articulo implements Serializable{
	private static final long serialVersionUID = 1L;
	private String calidad;
	private String tipo;
	private String especialidad;
    private  String tipoArticulo;
    private  String tipoVehiculo;
    private  double precio;
    private  String marca;
    public int cantidad; 
    private String marcaVehiculo;
    public long referencia;
    
    // Constructor
    public Articulo(String calidad,String tipo,String especialidad,String tipoArticulo, String tipoVehiculo, String marca, double precio, int cantidad,long referencia ) {
        this.calidad=calidad;
    	this.tipo=tipo;
    	this.tipoArticulo = tipoArticulo;
        this.especialidad = especialidad;
        this.tipoVehiculo = tipoVehiculo;
        this.precio = precio;
        this.marca = marca;
        this.cantidad = cantidad;
        this.referencia=referencia;
        
        if(tipo.equals("taller")) {
        	InventarioArticulo.articulos.add(this);
        	this.marcaVehiculo="Generico";
        }
        else if(tipo.equals("repuesto")) {
        	InventarioArticulo.repuestos.add(this);
        	this.marcaVehiculo="Generico";
        }
        
    }
    public Articulo(String calidad,String tipo,String especialidad,String tipoArticulo, String tipoVehiculo, String marca, double precio, int cantidad,String marcaVehiculo,long referencia ) {
        this.marcaVehiculo=marcaVehiculo;
    	this.calidad=calidad;
    	this.tipo=tipo;
    	this.tipoArticulo = tipoArticulo;
        this.especialidad = especialidad;
        this.tipoVehiculo = tipoVehiculo;
        this.precio = precio;
        this.marca = marca;
        this.cantidad = cantidad;
        this.referencia=referencia;
        
        if(tipo.equals("taller")) {
        	InventarioArticulo.articulos.add(this);
        }
        else if(tipo.equals("repuesto")) {
        	InventarioArticulo.repuestos.add(this);
        }
        
    }
    
    // Getters y Setters
    public String getTipoArticulo() {
        return tipoArticulo;
    }

    public void setTipoArticulo(String tipoArticulo) {
        this.tipoArticulo = tipoArticulo;
    }

    public String getTipoVehiculo() {
        return tipoVehiculo;
    }

    public void setTipoVehiculo(String tipoVehiculo) {
        this.tipoVehiculo = tipoVehiculo;
    }

    public double getPrecio() {
        return precio;
    }

    public void setPrecio(double precio) {
        this.precio = precio;
    }

    public String getMarca() {
        return marca;
    }

    public void setMarca(String marca) {
        this.marca = marca;
    }
   
	
	public String articulosDisponibles() {
		String informacion = tipoArticulo + " " + tipoVehiculo + " " + marca + " " + precio + "\n"  ;
		
		return informacion; 
	}

	public int getCantidad() {
		return cantidad;
	}

	public void setCantidad(int cantidad) {
		this.cantidad = cantidad;
	}

	public String getEspecialidad() {
		return especialidad;
	}

	public void setEspecialidad(String especialidad) {
		this.especialidad = especialidad;
	}
	public String info() {
		String texto = "Producto: " + getTipoArticulo() + "\n";
		return texto;
	}

	public String getTipo() {
		return tipo;
	}

	public void setTipo(String tipo) {
		this.tipo = tipo;
	}

	public String getCalidad() {
		return calidad;
	}

	public void setCalidad(String calidad) {
		this.calidad = calidad;
	}

	public String getMarcaVehiculo() {
		return marcaVehiculo;
	}

	public void setMarcaVehiculo(String marcaVehiculo) {
		this.marcaVehiculo = marcaVehiculo;
	}
	public void setReferencia(long referencia) {
		this.referencia=referencia;
	}
	public long getReferencia() {
		return referencia;
	}
	public static Articulo getArticuloPorReferencia(long referencia){
		Articulo finder = null;
		for(Articulo articulo: InventarioArticulo.articulos) {
			if(articulo.getReferencia() == referencia) {
				finder = articulo;
				break;
			}
		}
		return finder;
	}

	public static Articulo getRepuestoPorReferencia(long referencia){
		Articulo finder = null;
		for(Articulo articulo: InventarioArticulo.repuestos) {
			if(articulo.getReferencia() == referencia) {
				finder = articulo;
				break;
			}
		}
		return finder;
	}
	
	
   
    
}


    

File: PYTHON/Concesionario/gestorAplicacion/Activos/articulo.py
import sys
import os
ruta_activos = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', 'gestorAplicacion', 'Activos'))
ruta_personal = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', 'gestorAplicacion', 'Personal'))
ruta_gestor = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))
sys.path.append(ruta_activos)
sys.path.append(ruta_personal)
sys.path.append(ruta_gestor)
class Articulo:
    articulos = []
    repuestos = []

    def __init__(self, calidad, tipo, especialidad, tipoArticulo, tipoVehiculo, marca, precio, cantidad, referencia, marcaVehiculo="Generico"):
        self.marcaVehiculo = marcaVehiculo 
        self.calidad = calidad
        self.tipo = tipo
        self.tipoArticulo = tipoArticulo
        self.especialidad = especialidad
        self.tipoVehiculo = tipoVehiculo
        self.precio = precio
        self.marca = marca
        self.cantidad = cantidad
        self.referencia = referencia

        if tipo == "taller":
            Articulo.articulos.append(self)
            
        elif tipo == "repuesto":
            Articulo.repuestos.append(self)
            
    @classmethod
    def get_articulos(cls):
        return cls.articulos
    @classmethod
    def get_repuesto(cls):
        return cls.repuestos
    
    @classmethod
    def set_articulos(cls,Articulos):
        cls.articulos = Articulos

    @classmethod
    def set_repuestos(cls,repuestos):
        cls.repuestos = repuestos

    def set_tipoArticulo(self, tipoArticulo):
        self.tipoArticulo = tipoArticulo

    def get_tipoArticulo(self):
        return self.tipoArticulo

    def set_tipoVehiculo(self, tipoVehiculo):
        self.tipoVehiculo = tipoVehiculo

    def get_tipoVehiculo(self):
        return self.tipoVehiculo

    def set_precio(self, precio):
        self.precio = precio

    def get_precio(self):
        return self.precio

    def set_marca(self, marca):
        self.marca = marca

    def get_marca(self):
        return self.marca

    def articulos_disponibles(self):
        informacion = f"{self.tipoArticulo} {self.tipoVehiculo} {self.marca} {self.precio}\n"
        return informacion

    def set_cantidad(self, cantidad):
        self.cantidad = cantidad

    def get_cantidad(self):
        return self.cantidad

    def set_especialidad(self, especialidad):
        self.especialidad = especialidad

    def get_especialidad(self):
        return self.especialidad

    def info(self):
        texto = "Producto: " + self.get_tipoArticulo() + "\n"
        return texto

    def set_tipo(self, tipo):
        self.tipo = tipo

    def get_tipo(self):
        return self.tipo

    def set_calidad(self, calidad):
        self.calidad = calidad

    def get_calidad(self):
        return self.calidad

    def set_marcaVehiculo(self, marcaVehiculo):
        self.marcaVehiculo = marcaVehiculo

    def get_marcaVehiculo(self):
        return self.marcaVehiculo

    def set_referencia(self, referencia):
        self.referencia = referencia

    def get_referencia(self):
        return self.referencia

    @staticmethod
    def get_articuloPorReferencia(referencia):
        for articulo in Articulo.articulos:
            if articulo.get_referencia() == referencia:
                return articulo
        return None

    @staticmethod
    def get_repuestoPorReferencia(referencia):
        for articulo in Articulo.repuestos:
            if articulo.get_referencia() == referencia:
                return articulo
        return None
